core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -569,13 +571,17 @@ void postCommit() {
           }
 
           @Override
-          ClientStream newSubstream(ClientStreamTracer.Factory tracerFactory, Metadata newHeaders) {
-            CallOptions newOptions = callOptions.withStreamTracerFactory(tracerFactory);
+          ClientStream newSubstream(
+              Metadata newHeaders, ClientStreamTracer.Factory factory, boolean isTransparentRetry) {
+            CallOptions newOptions = callOptions;
+            newOptions = newOptions.withStreamTracerFactory(factory);
+            ClientStreamTracer[] tracers =
+                GrpcUtil.getClientStreamTracers(callOptions, isTransparentRetry);
```

Comment:
s/callOptions/newOptions/ ? Seems otherwise newOptions is useless.

Reply:
Hmmm. I'm surprised the test is not failing. I must add some regression test.

Reply:
Added regression test #8354